### ISSUE TITLE
can.view.attr should receive nodeLists

### ISSUE THREAD

**Issue by alexisabril** [canjs/canjs#2215]
This would allow safe DOM modification outside of a given element, $.fn.after, $.fn.wrap as an example

**Comment by dylanrtt** [comment]
I recently tried something like this:

```js
can.view.attr('my-attr', function(el, attrData) {
  var div = $('<div>');

  $(el).after(div);

  can.one.call(el, 'removed', function() {
    div.remove();
  });
});
```

Is that not safe? If it is, then what kind of DOM modification are you doing that requires nodeLists? Mostly just curious because I use custom attrs a lot.

**Comment by justinbmeyer** [comment]
The fix I did for #2267 adds nodeLists on `view.attr`

https://github.com/canjs/canjs/pull/2267/files#diff-d034a26ce8c4a4ae48f82780eca645cbR242

@dylanrtt What you are doing is likely fine. If your code looked like:

```
can.view.attr('my-attr', function(el, attrData) {
  var frag = can.stache("{{something}}")(attrData.scope);
  var livePart = frag.firstChild;
  $(el).after(frag);

  can.one.call(el, 'removed', function() {
    livePart.parentNode.removeChild(livePart);
  });
});
```

This wouldn't work because live text nodes without a parent element can't listen on a parent to know when to teardown their binding.

`$.fn.wrap` might also cause problems. But I'd have to experiment.  I think the nodeLists should still contain the right things even if the DOM is in the wrong order.

**Comment by justinbmeyer** [comment]
attr now gets node lists.
